src/main/java/com/gregtechceu/gtceu/api/machine/MetaMachine.java

```diff
@@ -132,48 +156,29 @@ public MetaMachine(IMachineBlockEntity holder) {
     //////////////////////////////////////
 
     @Override
-    public void markAsChanged() {
-        if (getHolder() instanceof ManagedSyncBlockEntity syncBlockEntity) {
-            syncBlockEntity.markAsChanged();
-        }
-    }
-
-    public @Nullable Level getLevel() {
-        return holder.level();
-    }
-
-    public BlockPos getPos() {
-        return holder.pos();
+    public final void setRemoved() {
+        super.setRemoved();
+        onUnload();
     }
 
-    public BlockState getBlockState() {
-        return holder.self().getBlockState();
+    public @UnknownNullability Level getLevel() {
+        return super.getLevel();
     }
 
     public void setOwnerUUID(UUID uuid) {
         ownerUUID = uuid;
         syncDataHolder.markClientSyncFieldDirty("ownerUUID");
     }
 
-    public boolean isRemote() {
-        return getLevel() == null ? GTCEu.isClientThread() : getLevel().isClientSide;
-    }
-
-    public void notifyBlockUpdate() {
-        holder.notifyBlockUpdate();
-    }
-
-    public void scheduleRenderUpdate() {
-        holder.scheduleRenderUpdate();
-    }
-
-    public void scheduleNeighborShapeUpdate() {
-        Level level = getLevel();
-        BlockPos pos = getPos();
-
-        if (level == null) return;
-
-        level.getBlockState(pos).updateNeighbourShapes(level, pos, Block.UPDATE_ALL);
+    @Override
+    public boolean triggerEvent(int id, int para) {
+        if (id == 1) { // chunk re render
```

Comment:
Is there an existing named constant anywhere for this?

Reply:
eventIDs are magic numbers throughout the MC code, there aren't constants for them